### ISSUE TITLE
Code review

### ISSUE THREAD

**Issue by olegakbarov** [GetExpert/redux-blog-example#4]
Would be cool if you guys let us know what do you think about this example and the ways of improving it :blush: 

cc @dan_abramov ;)

**Comment by vramana** [comment]
@olegakbarov Nice work, I think you should use https://github.com/github/fetch rather than `axios`.

**Comment by vramana** [comment]
It would be nice if you can explain the architecture in a bit more detail rather than working on new features.

**Comment by faergeek** [comment]
@vramana Hmmm, didn't know that fetch is not just for fetching data, but posting too.
Interesting, thanks.

**Comment by vramana** [comment]
What is advantage of using cookies vs something something like express-session ??

**Comment by olegakbarov** [comment]
We store a token in a cookie. The idea is to send a token with every call to API, to make sure we give right data for right user.This also provide benefits of scalability to platforms other then web, like iOS.

**Comment by faergeek** [comment]
Actually we use cookies because they're available on the client and on the server.

**Comment by cjroebuck** [comment]
I know you have the TODO comments already there but it would be interesting to see how you do the router redirects after a user authenticates or signs up [here] (https://github.com/GetExpert/redux-blog-example/blob/master/app/actions/auth.js#L49) and [here] (https://github.com/GetExpert/redux-blog-example/blob/master/app/actions/auth.js#L77). I'm currently using `componentWillReceiveProps` for this.

**Comment by vramana** [comment]
@olegakbarov Hey, I read a little about JWT. After reading a couple of article and a tutorial, I sort of understand them. But I still don't understand how to use them properly. Can you point to any good resources?

**Comment by olegakbarov** [comment]
@vramana take a look [here](https://stormpath.com/blog/where-to-store-your-jwts-cookies-vs-html5-web-storage) and [here](https://auth0.com/blog/2014/01/07/angularjs-authentication-with-cookies-vs-token/)

**Comment by vramana** [comment]
@olegakbarov Those are exactly the articles I read and tutorial from scotch.io. This repo implementation takes a slightly different approach than that. I am slightly confused at some places. Should I just re-read or do you have another examples I can look at?

**Comment by olegakbarov** [comment]
What questions do you have? Our example is extracted from the real-life project and may contain some redundant code. Also our `jsonServer` is just proof-of-concept.

**Comment by faergeek** [comment]
@cjroebuck We just thought to use it like action creator from component, because now we need to pass router to action creator.

Or we're going to try out redux-react-router by @acdlite which glues together redux and react-router better.

**Comment by victorkane** [comment]
Hi there. Thanks for a great example. Can you explain the npm start script, please: I don't understand the 'node .' segment, since there is no index.js in the root directory. Could you explain 

```
"scripts": {
    "start": "NODE_ENV=production babel-node ./api & node . & wait",
```
Thanks in advance!

**Comment by faergeek** [comment]
It starts script from `main` field in `package.json`.

**Comment by olegakbarov** [comment]
1. it sets NODE_ENV to "production"
2. executes content of `/api` with `babel-node`
3. executes content of current folder with good old `node` and sets it to foreground process

am i right @faergeek ?

**Comment by victorkane** [comment]
@olegakbarov Yes, but actually, the key is what @faergeek says, explaining the part I couldn't see, which is what `node .` means, and he explains that it starts the script from `main` field in `package.json`:

```
"main": "app/server-bundle",
```

Fantastic ... wonderful example, needs documentation so more can appreciate it. Thanks to both of you.

**Comment by faergeek** [comment]
@victorkane Sorry, I can't find documentation which describes exactly that behavior

@olegakbarov Not exactly.

1. It's right.
2. It executes `api/index.js` by default when you launch or require directory if there's no `package.json` in that directory (again, I'm not sure where it's described).
3. It executes two processes in background (`babel-node ./api` and `node .`) and `wait` is just a standard shell command (type `help wait` for more info) which waits when all of them stopped (it should never happen actually without Ctrl+C or some serious error :-)). It's somewhat like `Promise.all`.
